platform/android/export/export.cpp

```diff
@@ -1341,6 +1356,52 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			return str;
 		}
 	}
+
+	// Creates strings.xml files inside the gradle project for different locales.
+	void _create_project_name_strings_files(const Ref<EditorExportPreset> &p_preset) {
+		// TODO: should we use "package/name" or "application/config/name"?
+		String package_name = p_preset->get("package/name");
+		String project_name = get_project_name(package_name);
+		const String base_xml_string =
+				"<?xml version=\"1.0\" encoding=\"utf-8\"?>\n"
+				"<!--WARNING: THIS FILE WILL BE OVERWRITTEN AT BUILD TIME-->\n"
+				"<resources>\n"
+				"\t<string name=\"godot_project_name_string\">%s</string>\n"
+				"</resources>\n";
+		// Stores the string into the defualt values directory.
+		String processed_default_xml_string = vformat(base_xml_string, project_name.xml_escape(true));
+		store_string_at_path("res://android/build/res/values/godot_project_name_string.xml", processed_default_xml_string);
+
+		// Searches the Gradle project res/ directory to find all supported locales
+		DirAccessRef da = DirAccess::open("res://android/build/res");
+		if (!da) {
+			EditorNode::add_io_error("Could not open 'res://android/build/res' directory");
+			return;
+		}
+		da->list_dir_begin();
+		while (true) {
+			String file = da->get_next();
+			if (file == "") {
+				break;
+			}
+			if (!file.begins_with("values-")) {
```

Comment:
I forgot to mention that there are a couple other cases besides locales that may be in a "values-" prefixed folder, e.g. "values-v29" for SDK 29+ resources. Currently the only contents of [Godot's res directory](https://github.com/godotengine/godot/tree/master/platform/android/java/lib/res) are translated strings, but to make this a bit more future proof, could check that `locale` matches a "[a-z]{2}(-[A-Z]{2})?" type pattern. (I didn't test that regex.)

Alternatively, could just add a note/TODO saying that the code assumes all "values-" folders are locales.